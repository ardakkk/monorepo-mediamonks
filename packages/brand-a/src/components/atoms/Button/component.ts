import { Button, ButtonProps } from '@mediamonks/common';
import { colors } from 'core';

Button.defaultProps = {
  color: colors.variants.Text.White,
  backgroundColor: colors.variants.Element.Black,
};

export { Button, ButtonProps };
