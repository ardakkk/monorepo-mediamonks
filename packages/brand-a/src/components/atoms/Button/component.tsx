import styled from '@emotion/styled';
import type { FC } from 'react';
import React from 'react';

import type { Props } from './props';

const ButtonBase: FC<Props> = ({ children, ...rest }: Props) => (
  <button {...rest}>{children}</button>
);

export const Button = styled(ButtonBase)<Props>`
  background-color: #010103;
  color: #fff;
  border: none;
  padding: 10px 20px;
  border-radius: 7px;
`;
