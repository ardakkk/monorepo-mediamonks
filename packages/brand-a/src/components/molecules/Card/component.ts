import { Card, CardProps } from '@mediamonks/common';
import { colors } from 'core';

Card.defaultProps = {
  color: colors.variants.Element.Black,
};

export { Card, CardProps };
