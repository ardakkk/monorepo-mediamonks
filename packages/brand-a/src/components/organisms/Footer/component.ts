import { Footer, FooterProps } from '@mediamonks/common';
import { colors } from 'core';

Footer.defaultProps = {
  color: colors.variants.Text.White,
  backgroundColor: colors.variants.Brand.Black,
};

export { Footer, FooterProps };
