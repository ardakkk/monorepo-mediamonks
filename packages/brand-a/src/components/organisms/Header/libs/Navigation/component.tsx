import styled from '@emotion/styled';
import { colors } from 'core';
import type { FC } from 'react';
import React from 'react';

import type { Props } from './props';

const NavigationBase: FC<Props> = ({ items, ...rest }: Props) => (
  <nav {...rest}>
    <ul className="d-flex">
      {items?.length &&
        items.map((item, index) => (
          <li key={index} className="mr-2">
            <a href={item.href}>{item.label}</a>
          </li>
        ))}
    </ul>
  </nav>
);

export const Navigation = styled(NavigationBase)<Props>`
  a {
    text-decoration: none;
    color: ${colors.variants.Element.White};
  }
`;
