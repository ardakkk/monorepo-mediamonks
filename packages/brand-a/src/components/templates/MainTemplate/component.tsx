import { Footer, Header } from 'components/organisms';
import { colors } from 'core';
import type { FC } from 'react';
import React from 'react';

import type { Props } from './props';

export const MainTemplate: FC<Props> = ({ children }: Props) => (
  <div>
    <Header />
    <main>{children}</main>
    <Footer color={colors.variants.Text.White} backgroundColor={colors.variants.Brand.Black} />
  </div>
);
