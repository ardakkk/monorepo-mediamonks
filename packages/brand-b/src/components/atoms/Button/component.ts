import { Button, ButtonProps } from '@mediamonks/common';
import { colors } from 'core';

Button.defaultProps = {
  color: colors.variants.Text.White,
  backgroundColor: colors.variants.Element.Purple,
};

export { Button, ButtonProps };
