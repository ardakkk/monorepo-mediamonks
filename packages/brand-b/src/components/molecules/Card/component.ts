import { Card, CardProps } from '@mediamonks/common';
import { colors } from 'core';

Card.defaultProps = {
  color: colors.variants.Element.Purple,
};

export { Card, CardProps };
