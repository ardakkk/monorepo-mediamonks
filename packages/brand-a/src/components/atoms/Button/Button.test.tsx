import React from 'react';
import renderer from 'react-test-renderer';

import { Button } from './component';

test('Button component should be render', () => {
  const component = renderer.create(<Button>Click me</Button>);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
