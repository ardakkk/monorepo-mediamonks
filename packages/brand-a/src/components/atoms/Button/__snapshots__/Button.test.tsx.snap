// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should be render 1`] = `
<button
  className="css-1fk74k5"
  color="#fff"
  type="button"
>
  Click me
</button>
`;
